GNOME #438261; Debian #414698; fixes Thai input method when NumLock/CapsLock is
on

Index: modules/input/gtkimcontextthai.c
===================================================================
--- modules/input/gtkimcontextthai.c	(revision 17839)
+++ modules/input/gtkimcontextthai.c	(working copy)
@@ -288,7 +288,8 @@
   if (event->type != GDK_KEY_PRESS)
     return FALSE;
 
-  if (event->state & (GDK_MODIFIER_MASK & ~GDK_SHIFT_MASK) ||
+  if (event->state & (GDK_MODIFIER_MASK
+                      & ~(GDK_SHIFT_MASK | GDK_LOCK_MASK | GDK_MOD2_MASK)) ||
       is_context_lost_key (event->keyval))
     {
 #ifndef GTK_IM_CONTEXT_THAI_NO_FALLBACK
